lib/user.rb

```diff
@@ -0,0 +1,21 @@
+class User
+
+  attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(user, joke)
+    user.jokes << joke
```

Comment:
generally we would never want one object to *directly* write data into another object's attributes; how else could Sal tell Ali that joke so Ali now knows that joke?